tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_UpwindFlux.cpp

```diff
@@ -0,0 +1,226 @@
+                               spatial_metric_ext, d_spatial_metric_ext);
+  const auto pi_ext = GeneralizedHarmonic::pi(
+      lapse_ext, dt_lapse_ext, shift_ext, dt_shift_ext, spatial_metric_ext,
+      dt_spatial_metric_ext, phi_ext);
+
+  // More ingredients to get the char fields
+  const size_t n_pts = x.begin()->size();
+  const auto gamma_1 = make_with_value<Scalar<DataVector>>(x, 0.4);
+  const auto gamma_2 = make_with_value<Scalar<DataVector>>(x, 0.1);
+
+  // Get surface normal vectors
+  const DataVector one_over_one_form_magnitude_int =
+      1.0 / get(magnitude(
+                db::get<StrahlkorperTags::NormalOneForm<Frame::Inertial>>(box),
+                inverse_spatial_metric_int));
+  const auto unit_normal_one_form_int = StrahlkorperGr::unit_normal_one_form(
+      db::get<StrahlkorperTags::NormalOneForm<Frame::Inertial>>(box),
+      one_over_one_form_magnitude_int);
+  const auto unit_normal_vector_int = raise_or_lower_index(
+      unit_normal_one_form_int, inverse_spatial_metric_int);
+
+  // Get the characteristic fields and speeds
+  const auto char_fields_int = GeneralizedHarmonic::CharacteristicFieldsCompute<
+      spatial_dim, Frame::Inertial>::function(gamma_2,
+                                              inverse_spatial_metric_int,
+                                              spacetime_metric_int, pi_int,
+                                              phi_int,
+                                              unit_normal_one_form_int);
+  const auto char_fields_ext = GeneralizedHarmonic::CharacteristicFieldsCompute<
+      spatial_dim, Frame::Inertial>::function(gamma_2,
+                                              inverse_spatial_metric_ext,
+                                              spacetime_metric_ext, pi_ext,
+                                              phi_ext,
+                                              unit_normal_one_form_int);
+
+  const auto one = make_with_value<Scalar<DataVector>>(x, 1.0);
+  const auto minus_one = make_with_value<Scalar<DataVector>>(x, -1.0);
+  const auto five = make_with_value<Scalar<DataVector>>(x, 5.0);
+  const auto minus_five = make_with_value<Scalar<DataVector>>(x, -5.0);
+
+  std::array<DataVector, 4> char_speeds_one{
+      {get(one), get(one), get(one), get(one)}};
+  std::array<DataVector, 4> char_speeds_minus_one{
+      {get(minus_one), get(minus_one), get(minus_one), get(minus_one)}};
+  std::array<DataVector, 4> char_speeds_five{
+      {get(five), get(five), get(five), get(five)}};
+  std::array<DataVector, 4> char_speeds_minus_five{
+      {get(minus_five), get(minus_five), get(minus_five), get(minus_five)}};
+
+  GeneralizedHarmonic::UpwindFlux<spatial_dim> flux_computer{};
+
+  // If all the char speeds are +1, the weighted fields should just
+  // be the interior fields
+  const auto weighted_char_fields_one = flux_computer.weight_char_fields(
+      char_fields_int, char_speeds_one, char_fields_ext, char_speeds_one);
```

Comment:
I don't think you need the char fields to be computed from an actual solution for these tests, and could instead just generate random data for these, is that correct? Please try to remove all of the grid setup and analytic solution code, which is tested elsewhere and doesn't need to be tested here but makes up the majority of code in this test, and just `make_with_random_values` the `char_fields_int` and `char_fields_ext`.